### PR TITLE
Add jsloaderskipqueue

### PR DESCRIPTION
**Pull request by bebbi**
Per discussion in #417, this commit enables a caller to skip the queue.
This supports parallel loading of independent sequences of scripts

### CODE REVIEW

**Comment by bebbi** [comment]
(I'll add a test when the code is agreed)

**Comment by concavelenz** [comment]
Can you describe your use case and why you feel this is generally useful?

**Comment by bebbi** [comment]
This addresses an issue where it's impossible to parallelize the loading of independent sequence of scripts with multiple loadMany calls - e.g.: loadMany([A, B, C]) and loadMany([D, E, F]) can not be parallelized even if D/E/F have no dependency on A/B/C.

The proposed code allows skipping the queue forced by loadMany and have caller-side management of dependencies.

**Comment by bebbi** [comment]
Additionally, in terms of general usefulness, this commit allows a caller to manage their own loadMany queue however it suits them - i.e. this will make loadMany more generic and increase the number of possible use-cases it can support. Any feedback appreciated.

**Comment by joeltine** [comment]
Can you add a test for this please? Thanks!

**Comment by bebbi** [comment]
- added tests
- I couldn't reproduce the deps.js formatting using depswriter.py (License text etc), so I manually added the new files. Best practice feedback welcome.
- I noticed some capitalization issues in jsloader_test.js from my previous PR #417 . Do you prefer this commit to fix it or an addional one?

File: closure/goog/net/jsloader_test.js [hunk 1]
```diff
@@ -118,6 +118,22 @@ function testLoadMany() {
...
+    skipQueue: true
+  };
+  var skipQueueResult = goog.net.jsloader.loadMany(skipQueueTestUrls, options);
+
+  // Verify that the skipQueueResult is a separate deferred.
+  assertNotEquals('deferreds are not identical', skipQueueResult, result);
+
+  // Check that the independent test was not loaded yet.
+  assertEquals('verification object', null, window.test5);
+
```
Comment (joeltine) [inline closure/goog/net/jsloader_test.js]: Can you move these tests into their own `testLoadManySkipQueue` method?

**Comment by joeltine** [comment]
>I noticed some capitalization issues in jsloader_test.js from my previous PR #417 . Do you prefer this commit to fix it or an addional one?

Style/formatting changes should always be separate. This reduces required mental bandwidth of the reviewer, and protects those changes from being rolled back in the more complicated change.

File: closure/goog/net/jsloader_test.js (hunk 1 above)
Comment (bebbi) [inline closure/goog/net/jsloader_test.js]: Yes that should work. Unless you object, I'd still keep 1 loadMany + 1 loadMany(skipQueue) as the testing scenario though, it seems to make most sense.

**Comment by concavelenz** [comment]
I'm not yet convinced this is an appropriate addition.  Please don't merge this yet.

**Comment by bebbi** [comment]
@concavelenz I just run into another use-case now.
loading many scripts required:
loadMany(manyScriptURLs),
and in another caller:
loading 2 small scripts for more time critical stuff:
loadMany(smallstuff, {skipQueue: true})
Now there's a chance I can finish number 2 early without too much cross-talk, as long as the skipQueue feature exists. Otherwise my main view is going to wait long for all other stuff to load. Does that make sense?

**Comment by joeltine** [comment]
@concavelenz what are your concerns about this change? Being able to load discrete sets of dependencies in parallel seems like it could be useful. It's off by default, so there isn't much harm in enabling these use cases.

**Comment by bebbi** [comment]
Updated: separate test, and an additional commit for fixing comment syntax in merged PR #417

**Comment by bebbi** [comment]
any news on this?

**Comment by concavelenz** [comment]
The real question isn't whether the functionality is useful but whether it introduces unneeded complexity, confusion and risk to this class and if it would be better in a private fork of the class.   I'd feel better if others were advocating for the useful this of this change.

**Comment by bebbi** [comment]
I see. That's possibly an information dissemination problem then, I've tried discussing loadMany on closure google group before and have found helpful people but no user. github issues are probably a better place. Generally, would be good to establish a place to agree on features with owners before coding.

**Comment by joeltine** [comment]
>I've tried discussing loadMany on closure google group before and have found helpful people but no user.

I see you started [a thread on closure-library-discuss](https://groups.google.com/forum/#!searchin/closure-library-discuss/jsloader/closure-library-discuss/H-UI0bwVEsY/az3vxMF1L0gJ) back in January, which was the right thing to do. Closure has only been maintained by volunteers up until this June. These types of design proposals should get more attention going forward.

**Comment by joeltine** [comment]
Sorry @bebbi, just can't get consensus on this. I'd recommend you just use your own forked impl or overwrite loadMany in your code base.

**Pull request closed without merging**